src/actions/plugins/easymotion/easymotion.ts

```diff
@@ -240,25 +244,58 @@ export class EasyMotion {
           : this.getEasymotionMarkerForegroundColorOneChar();
 
       const renderOptions: vscode.ThemableDecorationInstanceRenderOptions = {
-        after: {
+        before: {
           contentText: keystroke,
           backgroundColor: this.getEasymotionMarkerBackgroundColor(),
-          height: `${configuration.easymotionMarkerHeight}px`,
-          width: `${keystroke.length * configuration.easymotionMarkerWidthPerChar}px`,
           color: fontColor,
-          textDecoration: `none;
+          width: `${keystroke.length * configuration.easymotionMarkerWidthPerChar}px;
+          margin: 0 ${configuration.easymotionMarkerMargin}px;
+          text-align: center;
+          border-radius: 1px;
           font-family: ${configuration.easymotionMarkerFontFamily};
           font-size: ${configuration.easymotionMarkerFontSize}px;
-          font-weight: ${configuration.easymotionMarkerFontWeight};
-          position: absolute;
-          z-index: 99;
-          bottom: ${configuration.easymotionMarkerYOffset}px`,
+          font-weight: ${configuration.easymotionMarkerFontWeight};`,
         },
       };
+
       // Position should be offsetted by the length of the keystroke to prevent hiding behind the gutter
       const charPos = pos.character + 1 + (keystroke.length - 1);
+      const range = new vscode.Range(pos.line, charPos, pos.line, charPos);
+
+      if (configuration.easymotionDimBackground) {
+        const dimPos =
+          pos.getLineEndIncludingEOL().character - charPos === 1 ? pos.character : charPos;
+
+        // This excludes markers from the dimming ranges by using them as anchors
+        // each marker adds the range between it and previous marker to the dimming zone
+        //
+        // example [m1] text that has multiple [m2] marks
+        // |<------    |<----------------------
+        if (dimmingZones.length === 0) {
+          dimmingZones.push(new vscode.Range(0, 0, pos.line, dimPos));
+        } else {
+          const prevDimPos = dimmingZones.slice(-1).pop();
```

Comment:
Can you change this to `dimmingZones[dimmingZones.length - 1]`? Then the `@ts-ignore`s below should be unnecessary

Reply:
yeah idk why the hell i did that way 😂